SimGeneral/TrackingAnalysis/plugins/TrackingTruthAccumulator.cc

```diff
@@ -1058,55 +1057,6 @@ namespace // Unnamed namespace for things only used in this file
 	}
 
 
-
-	//---------------------------------------------------------------------------------
-	//---------------------------------------------------------------------------------
-	//----   Functions in the unnamed namespace   -------------------------------------
-	//---------------------------------------------------------------------------------
-	//---------------------------------------------------------------------------------
-
-	int LayerFromDetid( const DetId& detId )
-	{
-		if( detId.det()!=DetId::Tracker ) return 0;
```

Comment:
The difference originates from this check, and what happens is the following. In the workflow 1306.0 the output from `LayerFromDetid` and `TrackerTopology::layer()` were different for the following DetId's
```
604049961
604049962
604049963
604049964
604049965
604049966
604050097
604050098
604050099
604050100
604050101
604050102
604050209
604050210
604050211
604050212
604050213
604050214
```
For these, the `DetId::det()` is equal to `DetId::Muon`. In line above, the returned layer is 0. However, in the new code, no such check is made, and `TrackerTopology::layer()` returns happily layer 1 (and assumes that the detectors are FPIX).

The best fix would probably be to make the check above before calling `TrackerTopology::layer()`. Another possibility coming to my mind would be for `TrackerTopology` to make these checks, but that would probably add too many typically redundant(?) checks to it.

By the way, is it documented somewhere that TrackerTopology assumes that the input `DetId`s belong to `DetId::Tracker`?